dask/dataframe/tests/test_multi.py

```diff
@@ -796,6 +796,23 @@ def test_concat_with_operation_remains_hlg():
     assert_eq(result, expected)
 
 
+def test_concat_dataframe_empty():
+    df = pd.DataFrame({"a": [100, 200, 300]}, dtype="int64")
+    empty_df = pd.DataFrame([], dtype="int64")
+    df_concat = pd.concat([df, empty_df])
+
+    ddf = dd.from_pandas(df, npartitions=1)
+    empty_ddf = dd.from_pandas(empty_df, npartitions=1)
+    ddf_concat = dd.concat([ddf, empty_ddf])
+    assert_eq(df_concat, ddf_concat)
+
+    empty_df_with_col = pd.DataFrame([], columns=["x"], dtype="int64")
+    df_concat_with_col = pd.concat([df, empty_df_with_col])
+    empty_ddf_with_col = dd.from_pandas(empty_df_with_col, npartitions=1)
+    ddf_concat_with_col = dd.concat([ddf, empty_ddf_with_col])
+    assert_eq(df_concat_with_col, ddf_concat_with_col)
```

Comment:
This suggestion makes the test pass, but there is a workaround due to subtleties explained above about `assert_eq` and how it check dtypes of the index. I tried to explain it into the note. 

I'm basically avoiding to check for dtypes in the assert and handling that separately. It's not the nicest but it works. what do we think? 

```suggestion
    # NOTE: check_dtypes compares dtypes of the index and for the case
    # of ddf_concat_with_col this is an Object, while the pandas version
    # df_concat_with_col is int64, pandas already knows the rows are empty while
    # dask is not aware. check_index=False doesn't avoid checking the dtype of
    # the index, hence the workaround below.
    assert_eq(df_concat_with_col, ddf_concat_with_col, check_dtype=False)

    assert (df_concat_with_col.index == ddf_concat_with_col.compute().index).all()
    assert df_concat_with_col.index.dtype == ddf_concat_with_col.compute().index.dtype
    assert (df_concat_with_col.dtypes == ddf_concat_with_col.dtypes).all()

```